pages/cloudflare/howtos/db.mdx

```diff
@@ -0,0 +1,232 @@
+import { cache } from "react";
+import * as schema from "./schema/d1";
+
+export const getDb = cache(() => {
+  const { env } = getCloudflareContext();
+  return drizzle(env.MY_D1, { schema });
+});
+
+// This is the one to use for static routes (i.e. ISR/SSG)
+export const getDbAsync = cache(async () => {
+  const { env } = await getCloudflareContext({ async: true });
+  return drizzle(env.MY_D1, { schema });
+});
+```
+
+#### Hyperdrive example
+
+```ts
+import { getCloudflareContext } from "@opennextjs/cloudflare";
+import { drizzle } from "drizzle-orm/node-postgres";
+import { cache } from "react";
+import * as schema from "./schema/pg";
+import { Pool } from "pg";
+
+export const getDb = cache(() => {
+  const { env } = getCloudflareContext();
+  const connectionString = env.HYPERDRIVE.connectionString;
+  const pool = new Pool({
+    // This env could be set either in the .env file or in wrangler.jsonc
+    connectionString: process.env.PG_URL,
+    // You don't want to reuse the same connection for multiple requests
+    maxUses: 1,
+  });
+  return drizzle({ client: pool, schema });
+});
+
+// This is the one to use for static routes (i.e. ISR/SSG)
+export const getDbAsync = cache(async () => {
+  const { env } = await getCloudflareContext({ async: true });
+  const connectionString = env.HYPERDRIVE.connectionString;
+  const pool = new Pool({
+    // This env could be set either in the .env file or in wrangler.jsonc
+    connectionString: process.env.PG_URL,
+    // You don't want to reuse the same connection for multiple requests
+    maxUses: 1,
+  });
+  return drizzle({ client: pool, schema });
+});
+```
+
+You can then use the `getDb` function to get a new client for each request. This will ensure that you don't run into any issues with connection pooling.
+
+## Prisma
+
+[Prisma](https://developers.cloudflare.com/d1/reference/community-projects/#prisma-orm) is a popular ORM for Node.js and TypeScript. It is designed to be easy to use and provides a lot of features out of the box. However, there are some subtleties to be aware of when using Prisma in Cloudflare Workers.
+
+### `schema.prisma`
+
+When using prisma in OpenNext, you do not want to provide an output directory for the generated client.
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+```
+
+This is because the generated client needs to be patched by OpenNext to work with Cloudflare Workers. If you provide an output directory, OpenNext will not be able to patch the client and it will not work.
+
+### `next.config.ts`
+
+Because prisma has some specific exports for cloudflare workers, you need to add the following to your `next.config.ts` file:
+
+```ts
+const nextConfig: NextConfig = {
+  serverExternalPackages: ["@prisma/client", ".prisma/client"],
+};
+```
+
+By doing this, this will ensure that both the generated client and the prisma client are included in the build for the correct runtime.
```

Comment:
```suggestion
By doing this, this will ensure that both the generated client and the prisma client are included in the build for the `workerd` runtime.
```